packages/trpc/server/routers/viewer/slots/util.ts

```diff
@@ -344,13 +345,69 @@ export async function getAvailableSlots({ input, ctx }: GetScheduleOptions) {
   }
   let currentSeats: CurrentSeats | undefined;
 
-  let usersWithCredentials = eventType.users.map((user) => ({
-    isFixed: !eventType.schedulingType || eventType.schedulingType === SchedulingType.COLLECTIVE,
-    ...user,
-  }));
-  // overwrite if it is a team event & hosts is set, otherwise keep using users.
-  if (eventType.schedulingType && !!eventType.hosts?.length) {
-    usersWithCredentials = eventType.hosts.map(({ isFixed, user }) => ({ isFixed, ...user }));
+  let usersWithCredentials;
+  let teamMember;
+
+  if (eventType.schedulingType === SchedulingType.ROUND_ROBIN && input.bookerEmail) {
+    let crmRoundRobinLeadSkip;
+    // See if CRM app is enabled and skip RR assignment
+    const eventTypeAppMetadata = eventType?.metadata?.apps;
+    for (const appKey in eventTypeAppMetadata) {
+      const app = eventTypeAppMetadata[appKey];
+      if (
+        app.enabled &&
+        app.appCategories &&
+        app.appCategories.some((category) => category === "crm") &&
+        app.roundRobinLeadSkip
+      ) {
+        crmRoundRobinLeadSkip = app;
+        break;
+      }
+    }
+
+    if (crmRoundRobinLeadSkip) {
+      const crmCredential = await prisma.credential.findUnique({
+        where: {
+          id: crmRoundRobinLeadSkip.credentialId,
+        },
+        include: {
+          user: {
+            select: {
+              email: true,
+            },
+          },
+        },
+      });
+      if (crmCredential) {
+        const crm = new CrmManager(crmCredential);
+        const contact = await crm.getContacts(input.bookerEmail, true);
+        if (contact) {
+          // Since this is enabled for round robin event, we iterate through hosts
+          const contactOwner = eventType.hosts.find((host) => host.user.email === contact[0].ownerEmail);
+          if (contactOwner) {
+            teamMember = contactOwner.user.email;
+            usersWithCredentials = [
+              contactOwner.user,
+              ...eventType.hosts.reduce((fixedHosts, host) => {
+                if (host.user.email !== contactOwner.user.email && host.isFixed) fixedHosts.push(host.user);
```

Comment:
What if the contactOwner is a fixed host. Then we still want to add the round robin hosts here, no?

Reply:
Good point. How do we feel about assigning the contact owner as a fixed host so the slots available are always available for the contact owner?
[`7bca279` (#14556)](https://github.com/calcom/cal.com/pull/14556/commits/7bca2794622d4e86f3fa906b4b5bfbfc0f3a1cd5)